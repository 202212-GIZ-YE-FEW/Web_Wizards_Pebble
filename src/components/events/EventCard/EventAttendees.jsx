const EventAttendees = ({ attendees, t, style }) => {
    return (
        <div style={{ display: "flex", gap: "6px" }}>
            <div style={{ marginInlineEnd: "10px" }}>
                {attendees?.slice(0, 3).map((item, index) => (
                    <a
                        href='#'
                        key={index}
                        className={
                            "!bg-black-100 !text-white rounded-full px-2 py-1 !transition-all " +
                            style
                        }
                    >
                        {item.charAt(0)}
                    </a>
                ))}
            </div>
            {attendees?.length > 0 ? (
                <span>
                    {" "}
                    + {attendees.length - 3} {t("attendees")}
                </span>
            ) : (
                <span>0 {t("attendees")}</span>
            )}
        </div>
    );
};

export default EventAttendees;
