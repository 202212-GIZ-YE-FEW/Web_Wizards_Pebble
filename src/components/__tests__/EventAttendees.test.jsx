import { I18nContext } from "next-i18next";
import renderer from "react-test-renderer";

import EventAttendees from "@/components/events/EventCard/EventAttendees";

it("renders correctly", () => {
    const i18n = {
        language: "en",
        t: (str) => str,
    };
    const tree = renderer
        .create(
            <I18nContext.Provider value={i18n}>
                <EventAttendees t={i18n.t} />
            </I18nContext.Provider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
