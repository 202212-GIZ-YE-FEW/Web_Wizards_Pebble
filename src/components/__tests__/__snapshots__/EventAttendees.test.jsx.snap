// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "gap": "6px",
    }
  }
>
  <div
    style={
      Object {
        "marginInlineEnd": "10px",
      }
    }
  />
  <span>
    0 
    attendees
  </span>
</div>
`;
